spec/booking_manager_spec.rb

```diff
@@ -0,0 +1,250 @@
+    end
+
+    it "increases the length of the reservations array by one" do
+      old_number = @hotel_rooms.reservations.length
+      new_booking = "new reservation"
+
+      @hotel_rooms.add_reservation(new_booking)
+      expect(@hotel_rooms.reservations.length).must_equal old_number + 1
+    end
+
+    it "adds new instance of Reservation to reservations array" do
+      room = @hotel_rooms.rooms[1]
+      another_booking = Hotel::Reservation.new(room, guest_name: "Polly Pocket", start_date: "May 10, 2018", end_date: "May 12, 2018")
+      @hotel_rooms.add_reservation(another_booking)
+
+      expect(@hotel_rooms.reservations.last).must_be_instance_of Hotel::Reservation
+      expect(@hotel_rooms.reservations.last.guest_name).must_equal "Polly Pocket"
+    end
+
+  end
+
+  describe "add_reservation_to_calendar method" do
+    it "creates a hash with room instance as key and dates as hash" do
+      hotel = Hotel::BookingManager.new(3)
+      # room = Hotel::Room.new(3)
+      room = hotel.rooms.first
+      another_booking = Hotel::Reservation.new(room, guest_name: "Kim Possible", start_date: "June 11, 2018", end_date: "June 14, 2018")
+
+      hotel.add_reservation_to_calendar(another_booking)
+      expect(hotel.add_reservation_to_calendar(another_booking)).must_be_kind_of Hash
+      expect(hotel.room_calendar[room]).must_include (Date.parse("June 11, 2018"))
+      expect(hotel.room_calendar[room].length).must_equal another_booking.number_nights
+    end
+  end
+
+  describe "get_reservation_cost method" do
+    it "returns the total cost of the reservation" do
+      hotel = Hotel::BookingManager.new(3)
+      room = hotel.rooms[3]
+      booking = Hotel::Reservation.new(room, guest_name: "Tony Tonson", start_date: "June 10, 2018", end_date: "June 12, 2018")
+
+      expect(hotel.get_reservation_cost(booking.cost_per_night, booking.number_nights)).must_be_close_to 400
+    end
+  end
+
+  describe "find_reservations_on_date method" do
+    it "returns all reservations on the desired date" do
+      hotel = Hotel::BookingManager.new(5)
+      room1 = hotel.rooms[0]
+      room2 = hotel.rooms[1]
+      room3 = hotel.rooms[2]
+      booking1 = Hotel::Reservation.new(room1, guest_name: "Marsha Daze", start_date: "June 10, 2018", end_date: "June 12, 2018")
+      booking2 = Hotel::Reservation.new(room2, guest_name: "Jessie Jade", start_date: "June 10, 2018", end_date: "June 14, 2018")
+      booking3 = Hotel::Reservation.new(room3, guest_name: "Daria Jade", start_date: "June 14, 2018", end_date: "June 16, 2018")
+      hotel.add_reservation_to_calendar(booking1)
+      hotel.add_reservation_to_calendar(booking2)
+      hotel.add_reservation_to_calendar(booking3)
+
+      expect(hotel.find_reservations_on_date("June 11, 2018", hotel.room_calendar).length).must_equal 2
+    end
+  end
+
+  describe "find_vacancies_on_date method" do
+    before do
+      @hotel = Hotel::BookingManager.new(5)
+      room1 = @hotel.rooms[0]
```

Comment:
There are a number of edge cases around when reservations overlap that you're missing here. A list was posted in slack at one point.